provider/aws_provider.go

```diff
@@ -7,9 +7,9 @@ type AwsProvider struct {
 func (a *AwsProvider) Endpoint() string {
 	var endpoint string
 	if a.region == "us-east-1" {
-		endpoint = "s3.amazonaws.com"
+		endpoint = "s3-fips.amazonaws.com"
```

Comment:
So this works fine, but there might be an even easier way.

You could update these lines:

https://github.com/cloud-gov/s3-broker/blob/ae28096bf7d8c5a71d62afc2af6ad304a9d34daa/broker/broker.go#L380-L381

to

```
				credentials.FIPSEndpoint = bucketDetails.FIPSEndpoint
				credentials.Endpoint = bucketDetails.FIPSEndpoint
```

Because there is already logic to determine the `FIPSEndpoint` over here:

https://github.com/cloud-gov/s3-broker/blob/ae28096bf7d8c5a71d62afc2af6ad304a9d34daa/awss3/s3_bucket.go#L271

Though it looks like that code might not handle the case where `region` is `us-east-1`

We could then probably remove all this `provider` code, but your PR doesn't need to do all that

Thoughts?

Reply:
Ultimately, it is up to you. It appears this was a fork at one point but is no longer. In that case, I am all for simplification and customization that meets our needs. This goes for cleaning up things we don't use like the Dockerfile as well.

Reply:
@spgreenberg Actually, `s3-fips.amazonaws.com` isn't a valid FIPS endpoint based on https://aws.amazon.com/compliance/fips/

So yeah, I think we should do the refactor proposed above and simplify the code

Reply:
@markdboyd I made these changes and dropped the provider. 

If this is no longer a fork, should we drop other unused references like minio and docker as well?

Reply:
@spgreenberg Sure, I'm fine to drop any code that you think is no longer necessary or relevant. But I won't hold up the PR on those changes